test/fields/array.test.ts

```diff
@@ -795,6 +795,243 @@ describe('buildFieldArray', () => {
+            },
+          },
+          required: ['children'],
+        }
+
+        const form = createHeadlessForm(schema, { initialValues: { children: [{ full_name: 'name' }] } })
+
+        expect(form.fields.find(({ name }) => name === 'children')?.fields?.find(({ name }) => name === 'field_1')?.isVisible).toBe(true)
+        expect(form.fields.find(({ name }) => name === 'children')?.fields?.find(({ name }) => name === 'field_2')?.isVisible).toBe(false)
+      })
+
+      it('correctly applies visibility when the condition is false', () => {
+        const schema: JsfObjectSchema = {
+          type: 'object',
+          additionalProperties: false,
+          properties: {
+            children: {
+              'items': {
+                'properties': {
+                  full_name: {
+                    'x-jsf-presentation': {
+                      inputType: 'text',
+                    },
+                    'title': 'Child Full Name',
+                    'type': 'string',
+                  },
+                  field_1: {
+                    'x-jsf-presentation': {
+                      inputType: 'text',
+                    },
+                    'title': 'Field 1',
+                    'type': 'string',
+                  },
+                  field_2: {
+                    'x-jsf-presentation': {
+                      inputType: 'text',
+                    },
+                    'title': 'Field 2',
+                    'type': 'string',
+                  },
+                },
+                'x-jsf-order': ['full_name', 'field_1', 'field_2'],
+                'required': ['full_name'],
+                'type': 'object',
+                'allOf': [
+                  {
+                    if: false,
+                    then: {
+                      required: ['field_1'],
+                      properties: {
+                        field_2: false,
+                      },
+                    },
+                    else: {
+                      required: ['field_2'],
+                      properties: {
+                        field_1: false,
+                      },
+                    },
+                  },
+                ],
+              },
+              'x-jsf-presentation': {
+                inputType: 'group-array',
+                addFieldText: 'Add new field',
+              },
+              'title': 'Children',
+              'type': 'array',
+            },
+          },
+          required: ['children'],
+        }
+
+        const form = createHeadlessForm(schema, { initialValues: { children: [{ full_name: 'name' }] } })
+
+        expect(form.fields.find(({ name }) => name === 'children')?.fields?.find(({ name }) => name === 'field_1')?.isVisible).toBe(false)
+        expect(form.fields.find(({ name }) => name === 'children')?.fields?.find(({ name }) => name === 'field_2')?.isVisible).toBe(true)
+      })
+    })
+
+    describe.skip('with logic based on answers', () => {
```

Comment:
I've created an [internal ticket](https://linear.app/remote/issue/RMT-1745/jsf-grouparray-conditional-fields-dont-work-visibility) for this